v22.1/oid.md

```diff
@@ -0,0 +1,19 @@
+---
+title: OID
+summary: The OID data type stores an unsigned 32 bit integer value.
+toc: true
+docs_area: reference.sql
+---
+
+The `OID` (Object Identifier) [data type](data-types.html) stores an unsigned 32 bit integer value. OIDs are used internally as primary keys for tables in [system schema](system-catalogs.html). OIDs are not used in user-created tables.
```

Comment:
sadly I think `OIDs` can be used in user-created tables.
```
create table baz (k oid primary key);
CREATE TABLE
```

Our OIDs are very similar to postgres's. There are "subtypes" of oids which correspond to different types. It's rather frustratingly complex. https://www.postgresql.org/docs/14/datatype-oid.html

One note is that pg_catalog is synthesized in cockroach whereas it's a real table in postgres. The structure of our OIDs is bespoke for each of the subtypes. We can omit that for now. I'm fine keeping this doc vague, but it is an area which could use further expansion.

To clarify what I mean, a table has a `regclass`. One can cast a string name to a regclass: 

```
CREATE TABLE foo (i INT);
SELECT 'foo'::regclass;
SELECT 'foo'::regclass::oid;
```

```
CREATE TABLE
  regclass
------------
  foo
(1 row)
  oid
-------
  116
(1 row)
```

Reply:
@ajwerner How do you want to move this forward?  Shall I just remove the sentence `OIDs are not used in user-created tables.`? What use cases should I document?

Reply:
I suggest something along the lines of `OIDs should not be used in user-created tables. Values of this type are not guaranteed to be stable across major releases`.

Reply:
i think we should also mention that users should not rely on OIDs to be globally unique. each OID subtype is only unique within certain namespace, and it probably isn't worth getting into all the details of that.